tests/controls/gridtest.cpp

```diff
@@ -742,4 +744,58 @@ void GridTestCase::ReadOnly()
 #endif
 }
 
+void GridTestCase::WindowAsEditorControl()
+{
+#if wxUSE_UIACTIONSIMULATOR
+    //The simple editor using wxWindow as a control
+    class TestEditor : public wxGridCellEditor
+    {
+    public:
+        TestEditor() = default;
+        TestEditor(const TestEditor&) = delete;
+        TestEditor& operator=(TestEditor const&) = delete;
+        void Create(wxWindow* parent, wxWindowID id, wxEvtHandler* evtHandler) override
+        {
+            SetWindow(new wxWindow(parent, id));
+            wxGridCellEditor::Create(parent, id, evtHandler);
+        }
+        void BeginEdit(int, int, wxGrid*) override {}
+        bool EndEdit(int, int, wxGrid const*, wxString const&, wxString* newval) override
+        {
+            *newval = GetValue();
+            return true;
+        }
+        void ApplyEdit(int row, int col, wxGrid* grid) override
+        {
+            grid->GetTable()->SetValue(row, col, GetValue());
+        }
+        void Reset() override {}
+        wxGridCellEditor* Clone() const override { return new TestEditor(); }
+        wxString GetValue() const override { return "value"; }
+    };
+
+    wxGridCellAttr* attr = new wxGridCellAttr();
+    attr->SetRenderer(new wxGridCellStringRenderer());
+    attr->SetEditor(new TestEditor());
+    m_grid->SetAttr(1, 1, attr);
+
+    EventCounter created(m_grid, wxEVT_GRID_EDITOR_CREATED);
+
+    wxUIActionSimulator sim;
+
+    m_grid->SetFocus();
+    m_grid->SetGridCursor(1, 1);
+    m_grid->EnableCellEditControl();
+
+    sim.Char(WXK_RETURN);
+
+    wxYield();
+
+    CPPUNIT_ASSERT_EQUAL(1, created.GetCount());
+
+    //Clean up
+    m_grid->SetAttr(1, 1, nullptr);
```

Comment:
If this is really required (I'm not sure if it is though), it should be done inside `wxON_BLOCK_EXIT_OBJ2` to ensure that it's called even if assert fails.